beater/journalbeat.go

```diff
@@ -43,6 +44,52 @@ type Journalbeat struct {
 	wg                 sync.WaitGroup
 }
 
+func (jb *Journalbeat) addMatchesForUnit(unit string) error {
+	var err error
+	AddMatch := func(s string) {
+		if err != nil {
+			return
+		}
+		err = jb.journal.AddMatch(s)
+	}
+
+	AddDisjunction := func() {
+		if err != nil {
+			return
+		}
+		err = jb.journal.AddDisjunction()
+	}
+
+	// Look for messages from the service itself
+	AddMatch("_SYSTEMD_UNIT=" + unit)
+
+	// Look for coredumps of the service
+	AddDisjunction()
+	AddMatch("MESSAGE_ID=fc2e22bc6ee647b6b90729ab34a250b1")
```

Comment:
@hookenz please reference the C sources with files / function names / line numbers from SystemD here for reference.

While it is clear to me (right now) that you copied the logic from `journalctl`, it would be good to get reminded of that while reading the code :)